### ISSUE TITLE
`filter_queryset` with invalid field causes UI crash on export

### ISSUE THREAD

**Issue by matthewhegarty** [django-import-export/django-import-export#1723]
**Describe the bug**

Run export with invalid field name via Admin UI and the crash screen is shown.

**To Reproduce**

Add the following to `BookResource` in `tests/core/admin.py`:

```python
    def filter_export(self, queryset, **kwargs):
        return queryset.select_related("x")
```

Then export books via the Admin UI.

**Versions (please complete the following information):**
 - Django Import Export: 4.0.0-beta
 - Python 3.8
 - Django 5.0

**Expected behavior**

It would be ideal to present a suitable error message to the user, instead of crashing the page.

**Screenshots**

![Screenshot 2023-12-20 at 15 28 19](https://github.com/django-import-export/django-import-export/assets/6249838/92a8a2ad-405b-4331-8d49-f944e08541f8)

**Comment by RobTilton** [comment]
is this a simple try block solution? Or are my instincts failing me?

**Comment by matthewhegarty** [comment]
Yes, it is because we are not catching `FieldError`.  This should be caught and returned to the user as an error message.  Similar issue to #1722.

**Comment by matthewhegarty** [comment]
This crash also occurs if you set either of the following to True and try an export:

```
IMPORT_EXPORT_SKIP_ADMIN_EXPORT_UI = True
IMPORT_EXPORT_SKIP_ADMIN_ACTION_EXPORT_UI = True
```
